### PR TITLE
No public setters on response model properties

### LINKED ISSUE #661: Convention test - response models should not use public setters for properties
Protected or private should be the only acceptable options.

### PR DESCRIPTION
**Pull request by khellang**
Fixes #661 :dancers:

### CODE REVIEW

**Comment by khellang** [comment]
Ugh. Now the `DebuggerDisplay` test fails. Have to fix that tomorrow :smile:

**Comment by shiftkey** [comment]
> Ugh. Now the `DebuggerDisplay` test fails.

LOL. So sorry!

**Comment by khellang** [comment]
> LOL. So sorry!

It's all fun :grin:

**Comment by khellang** [comment]
This PR got a bit out of hand. Once I started finding more models, the good old `DebuggerDisplay` test started failing as well. Anyway, now all model should have read-only properties and a `DebuggerDisplayAttribute` (with corresponding property) :grin:

File: Octokit.Tests.Conventions/ModelTests.cs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace Octokit.Tests.Conventions
+{
+    public class ModelTests
+    {
+        [Theory]
+        [MemberData("ModelTypes")]
+        public void AllModelsHasDebuggerDisplayAttribute(Type modelType)
```
Comment (shiftkey) [inline Octokit.Tests.Conventions/ModelTests.cs]: s/Has/Have

File: Octokit.Tests.Conventions/ModelTests.cs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+            var property = modelType.GetProperty("DebuggerDisplay", BindingFlags.Instance | BindingFlags.NonPublic);
+
+            Assert.NotNull(property);
+            Assert.Equal(typeof(string), property.PropertyType);
+        }
+
+        [Theory]
+        [MemberData("ResponseModelTypes")]
+        public void ResponseModelsHasGetterOnlyProperties(Type modelType)
```
Comment (shiftkey) [inline Octokit.Tests.Conventions/ModelTests.cs]: s/Has/Have

**Comment by shiftkey** [comment]
I'm happy with this cleanup - @haacked had a chance to look at it?

File: Octokit/Models/Response/SignatureResponse.cs [hunk 3]
```diff
@@ -7,16 +7,26 @@ namespace Octokit
     [DebuggerDisplay("{DebuggerDisplay,nq}")]
     public class SignatureResponse
     {
-        public string Name { get; set; }
-        public string Email { get; set; }
-        public DateTimeOffset Date { get; set; }
+        public SignatureResponse()
```
Comment (shiftkey) [inline Octokit/Models/Response/SignatureResponse.cs]: Not sure how I feel about adding in public default constructors. @haacked?

Comment (Haacked) [inline Octokit/Models/Response/SignatureResponse.cs]: For a `response` object, they might be necessary for the serializer to work.

Comment (shiftkey) [inline Octokit/Models/Response/SignatureResponse.cs]: Yes, that would do it :metal:

**Pull request merged**